docs/csharp/roslyn-sdk/compiler-api-model.md

```diff
@@ -0,0 +1,106 @@
+---
+title: .NET Compiler Platform SDK concepts and object model
+description: This overview provides the background you need to work effectively with the .NET compiler SDK. You'll learn the API layers, the major types involved, and the overall object model.
+author: billwagner
+ms.author: wiwagn
+ms.date: 10/10/2017
+ms.topic: conceptual
+ms.prod: .net
+ms.devlang: devlang-csharp
+ms.custom: mvc
+---
+# Understand the .NET Compiler Platform SDK model
+
+Compilers process the code you write following structured rules that often
+differ from the way humans read and understand code. A basic understanding
+of the model used by compilers is essential to understanding the APIs
+you use when building Roslyn-based tools. 
+
+## Compiler Pipeline Functional Areas
+
+The .NET Compiler Platform SDK exposes the C# and Visual Basic compilers' code analysis to you
+as a consumer by providing an API layer that mirrors a traditional compiler
+pipeline.
+
+![compiler pipeline](media/compiler-pipeline.png)
+
+Each phase of this pipeline is a separate component. First the
+parse phase tokenizes and parses source text into syntax that follows
+the language grammar. Second the declaration phase analyzes source and
+imported metadata to form named symbols. Next the bind phase matches identifiers
+in the code to symbols. Finally, the emit phase, emits an assembly with
+all the information built up by the compiler.
+
+![compiler pipeline api](media/compiler-pipeline-api.png)
```

Comment:
alt text should be a bit more descriptive. If you couldn't see the image, how would you describe that image?

Reply:
updated all image elements in this PR